pkg/deploy/tls-secrets.go

```diff
@@ -0,0 +1,247 @@
+
+	// ===== Check Che TLS certificate ===== //
+	cheTLSSecret := &corev1.Secret{}
+	err := clusterAPI.Client.Get(context.TODO(), types.NamespacedName{Namespace: checluster.Namespace, Name: cheTLSSecretName}, cheTLSSecret)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			// Error reading secret info
+			logrus.Errorf("Error getting Che TLS secert \"%s\": %v", cheTLSSecretName, err)
+			return reconcile.Result{RequeueAfter: time.Second}, err
+		}
+
+		// Che TLS secret doesn't exist, generate a new one
+
+		// Remove Che CA certificate secret if any
+		cheCASelfSignedCertificateSecret := &corev1.Secret{}
+		err = clusterAPI.Client.Get(context.TODO(), types.NamespacedName{Namespace: checluster.Namespace, Name: CheTLSSelfSignedCertificateSecretName}, cheCASelfSignedCertificateSecret)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				// Error reading secret info
+				logrus.Errorf("Error getting Che self-signed certificate secert \"%s\": %v", CheTLSSelfSignedCertificateSecretName, err)
+				return reconcile.Result{RequeueAfter: time.Second}, err
+			}
+			// Che CA certificate doesn't exists (that's expected at this point), do nothing
+		} else {
+			// Remove Che CA secret because Che TLS secret is missing (they should be generated together).
+			if err = clusterAPI.Client.Delete(context.TODO(), cheCASelfSignedCertificateSecret); err != nil {
+				logrus.Errorf("Error deleting Che self-signed certificate secret \"%s\": %v", CheTLSSelfSignedCertificateSecretName, err)
+				return reconcile.Result{RequeueAfter: time.Second}, err
+			}
+		}
+
+		// Prepare permissions for the certificate generation job
+		sa, err := SyncServiceAccountToCluster(checluster, CheTLSJobServiceAccountName, clusterAPI)
+		if sa == nil {
+			return reconcile.Result{RequeueAfter: time.Second}, err
+		}
+
+		role, err := SyncRoleToCluster(checluster, CheTLSJobRoleName, []string{"secrets"}, []string{"create"}, clusterAPI)
+		if role == nil {
+			return reconcile.Result{RequeueAfter: time.Second}, err
+		}
+
+		roleBiding, err := SyncRoleBindingToCluster(checluster, CheTLSJobRoleBindingName, CheTLSJobServiceAccountName, CheTLSJobRoleName, "Role", clusterAPI)
+		if roleBiding == nil {
+			return reconcile.Result{RequeueAfter: time.Second}, err
+		}
+
+		cheTLSSecretsCreationJobImage := DefaultCheTLSSecretsCreationJobImage()
+		jobEnvVars := map[string]string{
+			"DOMAIN":                         checluster.Spec.K8s.IngressDomain,
+			"CHE_NAMESPACE":                  checluster.Namespace,
+			"CHE_SERVER_TLS_SECRET_NAME":     cheTLSSecretName,
+			"CHE_CA_CERTIFICATE_SECRET_NAME": CheTLSSelfSignedCertificateSecretName,
+		}
+		job, err := SyncJobToCluster(checluster, CheTLSJobName, CheTlsJobComponentName, cheTLSSecretsCreationJobImage, CheTLSJobServiceAccountName, jobEnvVars, clusterAPI)
+		if err != nil || job == nil || job.Status.Succeeded == 0 {
+			logrus.Infof("Waiting on job '%s' to be finished", CheTLSJobName)
```

Comment:
I don't think we wait job when it is `nil` or an error happened...

Reply:
ok